catalyst/dl/experiments/runner.py

```diff
@@ -0,0 +1,177 @@
+from collections import OrderedDict
+from typing import Mapping, Any
+from abc import ABC, abstractmethod
+
+import torch
+from torch import nn
+
+from catalyst.dl.callbacks import Callback
+from catalyst.dl.state import RunnerState
+from catalyst.dl.utils import UtilsFactory
+from . import Experiment
+
+
+class ExperimentModelRunner(ABC):
+
+    def __init__(
+            self,
+            experiment: Experiment=None,
+            model: nn.Module=None,
+            device=None):
+        """
+
+        :param experiment: Experiment to run
+        :type experiment: Experiment
+        """
+        assert experiment or model
+
+        self.experiment = experiment
+        self.model: nn.Module = model
+        self.device = device
+
+        self.state: RunnerState = None
+        self.stage: str = None
+
+        self.callbacks: OrderedDict[str, Callback] = None
+
+        if device is None:
+            self._prepare_model()
+
+    @staticmethod
+    def _batch2device(batch: Mapping[str, Any], device):
```

Comment:
What about this design?
https://github.com/pytorch/pytorch/blob/master/torch/nn/parallel/scatter_gather.py#L11